core/middlewares/loader.js

```diff
@@ -0,0 +1,162 @@
+                        });
+                    }
+                }
+            }
+        });
+    }
+
+    return output;
+};
+
+var sortMiddlewares = function(groupsOrder, list){
+    var output = [];
+
+    if (!_.isArray(groupsOrder) && !list) return output;
+
+    var groupedList = {};
+
+    // Sort by groups
+    _.forOwn(list, function (value, key) {
+        var group = value.group || 'default';
+        var middleware = value;
+
+        if (!middleware.enabled) return;
+
+        middleware.name = middleware.name || key;
+
+        groupedList[group] = groupedList[group] || [];
+        groupedList[group].push(value);
+    });
+
+    // Sort each group
+    _.forOwn(groupedList, function (value, key) {
+        groupedList[key] = _.sortByOrder(value, ['order'], ['asc']);
+    });
+
+    // Concat groups by order
+    groupsOrder.forEach(function(item){
+        if (groupedList[item]) output = output.concat(groupedList[item]);
+    });
+
+    return output;
+};
+
+var loadMiddlewares = function(listArr, app){
+    if (!_.isArray(listArr) && !app) return;
+
+    log.debug('loading', listArr);
+
+    listArr.forEach(function(item){
+        if (item && fs.existsSync(item.indexPath)) {
+            app.use(require(item.indexPath).process);
+        }
+    });
+};
+
+module.exports.process = function(app, globalOptions){
+    var config = {
+        loadGroupsOrder: [
+            'request',
+            'pre-html',
+            'default',
+            'html',
+            'response'
+        ],
+        list: {
+            md: {
+                enabled: true,
+                order: -1,
+                group: 'pre-html',
+                indexPath: path.join(appRoot, 'core/middlewares/md.js')
+            },
+            mdTag: {
+                enabled: true,
+                order: 0,
+                group: 'html',
+                indexPath: path.join(appRoot, 'core/middlewares/mdTag.js')
+            },
+            clarify: {
+                enabled: true,
+                order: -2,
+                group: 'request',
+                indexPath: path.join(appRoot, 'core/middlewares/clarify.js')
+            },
+            read: {
+                enabled: true,
+                order: -1,
+                group: 'request',
+                indexPath: path.join(appRoot, 'core/middlewares/read.js')
+            },
+            send: {
+                enabled: true,
+                order: -1,
+                group: 'response',
+                indexPath: path.join(appRoot, 'core/middlewares/send.js')
+            },
+            wrap: {
+                enabled: true,
+                order: -1,
+                group: 'html',
+                indexPath: path.join(appRoot, 'core/middlewares/wrap.js')
+            }
+        }
+    };
+    utils.extendOptions(
+        config,
+        {
+            list: gatherMiddlewares(path.join(app.get('user'), 'node_modules'), new RegExp(/^sourcejs-/), 'core/middleware/index.js')
```

Comment:
I think `/^sourcejs-/` can be optional. And it would be nice to check it inside the `gatherMiddlewares` in this case.